### ISSUE TITLE
RegEx matching filter

### ISSUE THREAD

**Issue by robertn702** [supabase/pg_graphql#350]
# Summary
[summary]: #summary
Add ability to filter string values via [PostgreSQL RegEx matching](https://www.postgresql.org/docs/current/functions-matching.html#FUNCTIONS-POSIX-REGEXP).

# Rationale
[rationale]: #rationale

This would allow more advanced string filtering which `like`, `ilike`, and `startsWith` currently do not allow.

# Design
[design]: #design

I would propose that filters such as `match`, `imatch`, `nmatch`, `nimatch` be implemented which would map to `~`, `~*`, `!~`, `!~*`, respectively.

Hasura uses `_regex`, `_nregex`, `_iregex`, `_niregex`.
https://hasura.io/docs/latest/queries/postgres/query-filters/#text-search-or-pattern-matching-operators-_like-_similar-_regex-etc

# Examples
[examples]: #examples

The following filters would return an entries that contains the string "basketball" or "baseball" on the "text" column.
```json
{
  "filters": {
     "text": { 
        "imatch": "\b(basketball|baseball)"
     }
  }
}
```

# Drawbacks
[drawbacks]: #drawbacks

This would probably be a slow operation but provides functionality that would otherwise not be possible.

# Alternatives
[alternatives]: #alternatives

Partial functionality could be achieved with the addition of ["and", "or" where filters](https://github.com/supabase/pg_graphql/issues/87) along with existing "like" and "ilike" string filters. 

This alternative would not bring 100% feature parity with RegEx filtering and appears to be a more complex implementation.

# Unresolved Questions
[unresolved]: #unresolved-questions

None at this time.

**Comment by robertn702** [comment]
I'll give it a shot.

**Comment by olirice** [comment]
resolved by #351
